metaflow/plugins/metadata/service.py

```diff
@@ -19,9 +28,15 @@ def __init__(self, msg, http_code=None, body=None):
 
 class ServiceMetadataProvider(MetadataProvider):
     TYPE = 'service'
+    HB_URL_KEY = 'hb_url'
 
     def __init__(self, environment, flow, event_logger, monitor):
         super(ServiceMetadataProvider, self).__init__(environment, flow, event_logger, monitor)
+        self.mli_url_task_template = METADATA_SERVICE_URL + \
```

Comment:
Can we use `urlparse` to construct the URL so that trailing slashes in `METADATA_SERVICE_URL` are appropriately taken care of?

Reply:
used urljoin from urlparse
note that it requires the user specify the protocol (http vs https) otherwise it won't work
```
>>> urljoin('http://www.example.com:8080//', '/flows/{flow_id}/runs/{run_number}/steps/{step_name}/tasks/{task_id}/heartbeat')
'http://www.example.com:8080/flows/{flow_id}/runs/{run_number}/steps/{step_name}/tasks/{task_id}/heartbeat'
>>> urljoin('www.example.com:8080//', '/flows/{flow_id}/runs/{run_number}/steps/{step_name}/tasks/{task_id}/heartbeat')
'/flows/{flow_id}/runs/{run_number}/steps/{step_name}/tasks/{task_id}/heartbeat'
```